pkg/tracing/logger.go

```diff
@@ -50,21 +53,37 @@ func NewLogger(logger *zap.Logger, tracer trace.Tracer, attrs ...attribute.KeyVa
 }
 
 func (l traceLogger) StartSpan(ctx context.Context, op string, attrs ...attribute.KeyValue) (context.Context, ScopedLogger) {
+	childCtx, childSpan := l.Tracer.Start(ctx, op)
+
+	l.addAttributesToSpan(childCtx, childSpan, attrs...)
+
+	return NewScopedLogger(childCtx, childSpan, l)
+}
+
+func (l traceLogger) StartSpanNamedByCaller(ctx context.Context, attrs ...attribute.KeyValue) (context.Context, ScopedLogger) {
```

Comment:
The implementation for this function looks like it could be reduced down to the following, no? Though I'm still trying to figure out when someone would use StartSpan vs StartSpanByCaller?

```
return  l.StartSpan(ctx, callerFunc(0), attrs...)
```